pkg/controllers/webhook/controller.go

```diff
@@ -650,9 +655,12 @@ func (c *controller) buildResourceMutatingWebhookConfiguration(ctx context.Conte
 					} else {
 						c.mergeWebhook(fail, p, false)
 					}
+					rules := p.GetSpec().Rules
+					operationStatusMap = computeOperationsForMutatingWebhookConf(rules, operationStatusMap)
```

Comment:
Shouldn't operations be associated with the corresponding matching resources?

Given this rule:
```
  rules:
    - name: rule-1
      match:
        resources:
          kinds:
            - Namespace
          operations:
          - CREATE
...
    - name: rule-2
      match:
        resources:
          kinds:
            - Deployment
          operations:
          - UPDATE
```
It should be converted to:
```
webhooks:
- rules:
  - apiGroups:
    - ''
    apiVersions:
    - 'v1'
    operations:
    - CREATE
    resources:
    - 'namespaces'
    scope: '*'
- rules:
  - apiGroups:
    - 'apps'
    apiVersions:
    - 'v1'
    operations:
    - UPDATE
    resources:
    - 'deployments'
    scope: '*'
```
```

Reply:
Right, earlier we would have something like this
```
  rules:
  - apiGroups:
    - ""
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    - DELETE
    - CONNECT
    resources:
    - namespaces
    scope: '*'
  - apiGroups:
    - apps
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    - DELETE
    - CONNECT
    resources:
    - deployments
    scope: '*'
  sideEffects: NoneOnDryRun
  timeoutSeconds: 10
```

But now we would have the minimum subset applied to all, like this:
```
  rules:
  - apiGroups:
    - ""
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    resources:
    - namespaces
    scope: '*'
  - apiGroups:
    - apps
    apiVersions:
    - v1
    operations:
    - CREATE
    - UPDATE
    resources:
    - deployments
    scope: '*'
```

Let me try to make it corresponding to matching resources